### ISSUE TITLE
Kafka in Azure Event Hub

### ISSUE THREAD

**Issue by sleipnir** [akkadotnet/Akka.Streams.Kafka#148]
I'm trying to find some example of using Akka.NET Streams to consume Kafka events from the Azure Event Hub. It seems to me that Kafka support does not support the authentication required to connect to Azure Hub, but I may be mistaken. If anyone has an example I would appreciate it very much.

I'm not actually a .Net developer by nature. In Java or Scala using Akka Streams I would configure my application like this:

```
akka {
  kafka.producer {
    #Akka kafka producer properties can be defined here
    client.id = "proxy-client"


    # Properties defined by org.apache.kafka.clients.producer.ProducerConfig
    # can be defined in this configuration section.
    kafka-clients {
      bootstrap.servers = "my-kafka.servicebus.windows.net:9093"
      sasl.mechanism = PLAIN
      security.protocol = SASL_SSL
      sasl.jaas.config = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"$ConnectionString\" password=\"Endpoint=sb://my-kafka.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey={key}\";"
    }
  }
```
On the Gitter channel, Aaron told me that you support these options so it must be the case that you just find them. So an example would be welcome.

**Comment by IgorFedchenko** [comment]
@sleipnir Indeed, underlying [Confluent Kafka driver](https://github.com/confluentinc/confluent-kafka-dotnet) supports all of that options - you can see the full list here: https://docs.confluent.io/current/installation/configuration/producer-configs.html#cp-config-producer . Just search for "sasl.mechanism" and others on the page.

To configure this settings, you can set them in code like this on producer configuration:
```
var producerSettings = ProducerSettings<Null, string>.Create(system, null, null)
    .WithBootstrapServers("localhost:9092")
    .WithProperty("sasl.mechanism", "PLAIN");
```

I think we do not support HOCON configuration for such things at the moment, so you will need to set all of this in code. But you can use your own HOCON configuration to load this values from file, if you want.

So the idea is to set the settings you need like in the sample above, and they will be passed to internal Confluent Kafka client instance.

Hope this helps!
